File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.cs
var input = Array.ConvertAll(Console.ReadLine().Split(), int.Parse);

var edge = Enumerable.Range(0, input[0]).Select(x => new List<int>()).ToList();

for (int i = 0; i < input[1]; i++)
{
    var get = Array.ConvertAll(Console.ReadLine().Split(), int.Parse);
    edge[get[0] - 1].Add(get[1] - 1);
    edge[get[1] - 1].Add(get[0] - 1);
}

for (int i = 0; i < edge.Count; i++) edge[i].Sort();

Console.WriteLine(string.Join(" ", dfs(edge)));
Console.WriteLine(string.Join(" ", bfs(edge)));

List<int> bfs(List<List<int>> e)
{
    Queue<int> que = new Queue<int>();
    var log = new List<int>();
    var visited = new bool[input[0]];

    que.Enqueue(input[2] - 1);
    visited[input[2] - 1] = true;

    while (que.Count > 0)
    {
        var now = que.Dequeue();
        if(!log.Contains(now + 1)) log.Add(now + 1);

        foreach (var next in edge[now])
        {
            if (!visited[next])
            {
                visited[now] = true;
                que.Enqueue(next);
            }
        }
    }

    return log;
}

List<int> dfs(List<List<int>> e)
{
    Stack<int> st = new Stack<int>();
    var log = new List<int>();
    var visited = new bool[input[0]];

    st.Push(input[2] - 1);
    visited[input[2] - 1] = true;

    while (st.Count > 0)
    {
        var now = st.Pop();
        visited[now] = true;
        if (!log.Contains(now + 1)) log.Add(now + 1);

        foreach (var next in edge[now].AsEnumerable().Reverse())
        {
            if (!visited[next])
            {
                st.Push(next);
            }
        }
    }

    return log;
}

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.cc
#include <iostream>
#include <vector>
#include <queue>
using namespace std;

queue<int> que;
vector<vector<int>> edgeList;
vector<bool> visited;
void dfs(int);
void bfs(int);

int main()
{
    ios::sync_with_stdio(false); cin.tie(NULL); cout.tie(NULL);

    int n, m, src;
    cin >> n >> m >> src;

    edgeList.resize(n + 1);

    for(int i = 0; i < m; i++){
        int s, e;
        cin >> s >> e;
        edgeList[s].push_back(e);
        edgeList[e].push_back(s);
    }

    for(auto& edge : edgeList){
        sort(edge.begin(), edge.end());
    }

    visited = vector<bool>(n + 1, false);
    dfs(src);
    cout << '\n';
    visited = vector<bool>(n + 1, false);
    bfs(src);
    cout << '\n';
}

void dfs(int node){
    cout << node << ' ';
    visited[node] = true;
    for(auto v : edgeList[node]){
        if(!visited[v]){
            dfs(v);
        }
    }
}

void bfs(int node){
    que.push(node);
    visited[node] = true;

    while(!que.empty()){
        int _node = que.front();
        cout << _node << ' ';
        que.pop();

        for(auto v : edgeList[_node]){
            if(!visited[v]){
                visited[v] = true;
                que.push(v);
            }
        }
    }
}